src/Raven.Server/Documents/Queries/StreamJsonlDocumentQueryResultWriter.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Session;
+using Raven.Server.Json;
+using Sparrow.Extensions;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Queries
+{
+    public class StreamJsonlDocumentQueryResultWriter : IStreamQueryResultWriter<Document>
+    {
+        private readonly AsyncBlittableJsonTextWriter _writer;
+        private readonly JsonOperationContext _context;
+
+        public StreamJsonlDocumentQueryResultWriter(Stream stream, JsonOperationContext context)
+        {
+            _context = context;
+            _writer = new AsyncBlittableJsonTextWriter(context, stream);
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            return _writer.DisposeAsync();
+        }
+
+        public void StartResponse()
+        {
+        }
+
+        public void StartResults()
+        {
+        }
+
+        public void EndResults()
+        {
+        }
+
+        public async ValueTask AddResultAsync(Document res, CancellationToken token)
+        {
+            _writer.WriteStartObject();
+            _writer.WritePropertyName("Item");
+            _writer.WriteDocument(_context, res, metadataOnly: false);
+            _writer.WriteEndObject();
+
+            _writer.WriteNewLine();
+            await _writer.MaybeFlushAsync(token);
+        }
+
+        public void EndResponse()
+        {
+        }
+
+        public ValueTask WriteErrorAsync(Exception e)
+        {
+            _writer.WriteStartObject();
+            _writer.WritePropertyName("Error");
+            _writer.WriteString(e.ToString());
+            _writer.WriteEndObject();
+
+            _writer.WriteNewLine();
+
+            return default;
+        }
+
+        public ValueTask WriteErrorAsync(string error)
+        {
+            _writer.WriteStartObject();
+            _writer.WritePropertyName("Error");
+            _writer.WriteString(error);
+            _writer.WriteEndObject();
+
+            _writer.WriteNewLine();
+
```

Comment:
here we probably want to call `Flush` directly, to ensure that this is sent immediately to the user.

Reply:
Add a FlushAsync below, let me know if that's what you wanted.

Reply:
yes, exactly.

Will you also be doing the PR on the `node.js` client?

Reply:
Would love to. I can update RDBC-622 with my progress, or lack of :)

Reply:
Thanks, that would be great